content/blog/improved-preview-experience/index.md

```diff
@@ -0,0 +1,77 @@
+Kubernetes cluster.
+
+{{< gist pulumipus 0c6e4ede557600e114885869ea0aeb96 >}}
+
+A simple change to the labels for the GKE cluster led to a preview that suggested that the dependent resources would need to be replaced.
+
+```
+     Type                              Name                         Plan        Info
+     pulumi:pulumi:Stack               gcp-ts-gke-hello-world-dev2
+ ~   ├─ gcp:container:Cluster          helloworld                   update      [diff: ~resourceLabels]
+ +-  ├─ gcp:container:NodePool         primary-node-pool            replace     [diff: ~location]
+ +-  ├─ pulumi:providers:kubernetes    helloworld                   replace     [diff: ~kubeconfig]
+ +-  ├─ kubernetes:core/v1:Namespace   helloworld                   replace     [diff: -metadata~provider]
+ +-  ├─ kubernetes:core/v1:Service     helloworld                   replace     [diff: ~metadata,provider]
+ +-  └─ kubernetes:apps/v1:Deployment  helloworld                   replace     [diff: ~metadata,provider]
+
+Resources:
+    ~ 1 to update
+    +-5 to replace
+    6 changes. 1 unchanged
+```
+
+The result of the Pulumi apply showed that only the `gcp:container:Cluster` changed, and no other Kubernetes resources were replaced. This is because Pulumi
+was trying to be conservative and let you know what "could" happen. It was calculating that changes to the cluster could potentiall lead to the outputs changing
+which other resource rely on. In practice, that wasn't going to happen but Pulumi didn't previously have a way to check for that and and had to be conservative
+to be correct 
+
+```
+     Type                      Name                         Status      Info
+     pulumi:pulumi:Stack       gcp-ts-gke-hello-world-dev2
+ ~   └─ gcp:container:Cluster  helloworld                   updated     [diff: ~resourceLabels]
+
+Resources:
+    ~ 1 updated
+    6 unchanged
+```
+
+Pulumi now supports asking the individual resource providers about what outputs will change as part of an update to a resource.
+This allows the providers (like the GCP provider in this case) to give more specific information, which allows Pulumi to provide more accurate previews:
+
+```
+     Type                      Name                         Plan       Info
+     pulumi:pulumi:Stack       gcp-ts-gke-hello-world-dev2
+ ~   └─ gcp:container:Cluster  helloworld                   update     [diff: ~resourceLabels]
+
+Resources:
+    ~ 1 to update
+    6 unchanged
+```
+
+## Conclusion
+
+The improved preview is available in Pulumi v2.12.0 and is supported in all new provider releases.
+If you would like to disable this new behaviour and preserve the previous, more conservative behaviour, you can set an
+environment variable `PULUMI_DISABLE_PROVIDER_PREVIEW` to a truthy value (`1` or `true`).
+If you find any problems, [please do let us know](https://github.com/pulumi/pulumi/issues/new).
```

Comment:
I'd leave this last sentence off as it's generally true for everything we produce and I'd rather have preview issues show up in the provider-specific repo to first have a pass at if it's specific to the provider vs. a general issue.